### PR TITLE
detect fails when mask doesn't cover inetnum

### PR DESCRIPTION
**Pull request by petchells**
compares mask against bits in the inetnum and fails if any aren't covered

### CODE REVIEW

**Comment by eshryane** [comment]
Looks good to me

**Comment by eshryane** [comment]
Don't merge yet - a test is failing in DomainObjectServiceTestIntegration:
create_domain_object_unparsable_json

**Comment by eshryane** [comment]
There are also failing Groovy tests (*Spec.groovy in whois-endtoend).

**Comment by petchells** [comment]
All tests have been fixed: http://bamboo.ripe.net/browse/DB-WHOIS119

**Comment by eshryane** [comment]
Reviewing now

File: whois-query/src/test/java/net/ripe/db/whois/query/support/QueryExecutor.java [hunk 1]
```diff
@@ -31,12 +30,10 @@
     private static final Set<String> ERRORS_TO_IGNORE = Sets.newHashSet("ERROR:101:", "ERROR:106:");
 
     private final QueryExecutorConfiguration configuration;
-    private final AccessControlListManager accessControlListManager;
     private final Logger logger;
 
-    public QueryExecutor(final QueryExecutorConfiguration configuration, final AccessControlListManager accessControlListManager, final Logger logger) throws UnknownHostException {
```
Comment (eshryane) [inline whois-query/src/test/java/net/ripe/db/whois/query/support/QueryExecutor.java]: Confirmed aclManager is not needed in test support class - personal object accounting is done with hardcoded object type check.

**Comment by agoston** [comment]
Haha, we had the exact same issue years back. Turns out netmasks like 10.0.0.1/8 are perfectly valid as per RFC. Hence we left this in there intentionally.

**Comment by agoston** [comment]
It's called combined host and network addresses, and is already used in RIPE DB. I wouldn't go and implement this without announcement, it's actually a handy feature to be able to specify a specific host inside the network.

Another issue is that it's implemented in IPV4Resource, so it pretty much applies to _every_ attribute. Note, this is a major change, even if it doesn't seem so. Announcement will definitely be necessary if you're planning to go through with this.

**Pull request merged**